src/panels/lovelace/entity-rows/hui-weather-entity-row.ts

```diff
@@ -141,30 +124,25 @@ class HuiWeatherEntityRow extends LitElement implements LovelaceRow {
       return undefined;
     }
 
-    const forecastNow = stateObj.attributes.forecast[0];
+    const extrema = this._getExtrema(stateObj);
+
+    if (extrema) {
+      return extrema;
+    }
 
     let value: string;
     let attribute: string;
 
-    if (forecastNow.precipitation) {
-      value = forecastNow.precipitation.toString();
+    if ("precipitation" in stateObj.attributes.forecast[0]) {
+      value = stateObj.attributes.forecast[0].precipitation!.toString();
```

Comment:
The template will make it a string
```suggestion
      value = stateObj.attributes.forecast[0].precipitation;
```